yaml/src/Data/Yaml/Pretty.hs

```diff
@@ -62,14 +69,20 @@ defConfig = Config mempty False
 
 -- |
 -- @since 0.8.13
-getConfCompare :: Config -> Text -> Text -> Ordering
+getConfCompare :: Config -> [Pos] -> Text -> Text -> Ordering
```

Comment:
Hmm... this will result in a breaking API change, but it looks minor (and unavoidable). Any thoughts on whether some kind of a workaround (like you have with `setConfComparePos`) would be possible?

Reply:
I think it is impossible. I made `confCompare` value bigger, you can't fit bigger value into smaller without some complicated encoding. Granted, `Text ~ Nat` as far as set theory concerned, but that is not helpful from practical point of view.

Another approach is that I can pick some arbitrary value of type `[Pos]` and apply it. That will preserve signature, but will violate get-set expectations.

Reply:
BTW, just learned about JSONPath (https://hackage.haskell.org/package/aeson-2.1.1.0/docs/Data-Aeson-Internal.html#t:JSONPath). Probably should re-use it.